### PR TITLE
[EBPF] gpu: do not store symbols that aren't used for CUDA kernels

### PR DESCRIPTION
**Pull request by gjulianm**
### What does this PR do?

Filters the symbols that we keep track of in the `GetSymbols` function to avoid increasing memory usage unnecesarily. We only need symbols for CUDA kernels.

This was causing some extra memory usage in the symbol data being stored, bigger in cases where the binary had more symbols. After this change, the memory usage is negligible as it doesn't show on profiles or shows only a handful of bytes.

### Motivation

Reduce memory usage seen in load-test environment and local profiling.

### Describe how to test/QA your changes

Tested via unit tests/e2e tests. Also tested manually.

### CODE REVIEW

File: pkg/gpu/cuda/fatbin.go [hunk 1]
```diff
@@ -52,6 +52,16 @@ func (fb *Fatbin) GetKernel(name string, smVersion uint32) *CubinKernel {
 	return fb.Kernels[key]
 }
 
+// HasKernelWithName returns true if the fatbin has a kernel with the given name
+func (fb *Fatbin) HasKernelWithName(name string) bool {
+	for key := range fb.Kernels {
+		if key.Name == name {
```
Comment (brycekahle) [inline pkg/gpu/cuda/fatbin.go:58]: Can you create a `map[string]struct{}` ahead of time for the kernel names? That way you don't have to do this loop for each symbol.

Comment (gjulianm) [reply pkg/gpu/cuda/fatbin.go:58]: Done!

**Review by brycekahle — APPROVED** [review]

**Pull request merged**